modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -324,7 +326,15 @@ protected void initChannel(Channel ch) throws Exception {
             ch.pipeline()
                 .addLast("decoder", decoder)
                 .addLast("decoder_compress", new HttpContentDecompressor())
-                .addLast("encoder", new HttpResponseEncoder())
+                .addLast("encoder", new HttpResponseEncoder() {
+                    @Override
+                    protected boolean isContentAlwaysEmpty(HttpResponse msg) {
+                        // non-chunked responses (Netty4HttpResponse extends Netty's DefaultFullHttpResponse) with chunked transfer
+                        // encoding are only sent by us in response to HEAD requests an must always have an empty body
+                        return msg instanceof Netty4HttpResponse && HttpUtil.isTransferEncodingChunked(msg)
```

Comment:
It's a little dirty to guess things like this, but it seemed nicer than unnecessarily adding another field to the netty response implementation.

Reply:
Seems _ok_ but would it also work to just check if there's anything in the body (i.e. `((Netty4HttpResponse)msg).content().isReadable()`)? At least IMO we should assert there's no body here.